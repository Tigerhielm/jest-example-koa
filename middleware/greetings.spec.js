const greetings = require('./greetings');

test('greetings works', async () => {
  const ctx = {};

  await greetings(ctx, () => {});

  expect(ctx.body).toBe('Hello. Remember to subscribe.');
});

test('greetings works before-and-after', async () => {
  const ctx = {};
  const next = jest.fn(() => {
    expect(ctx.body).toBe('Hello.');
    ctx.body += ' I am content.';
  });

  await greetings(ctx, next);

  expect(next).toHaveBeenCalledTimes(1);
  expect(ctx.body).toBe('Hello. I am content. Remember to subscribe.');
});

test('greetings works complete', async () => {
  const ctx = {
    response: { set: jest.fn() }
    /* ADD OTHER MOCKS */
  };
  const next = jest.fn(() => {
    expect(ctx).toMatchSnapshot();
  });

  await expect(greetings(ctx, next)).resolves.toBeUndefined();

  expect(next).toHaveBeenCalledTimes(1);
  expect(ctx).toMatchSnapshot();
  expect(ctx.response.set.mock.calls).toMatchSnapshot();
});
