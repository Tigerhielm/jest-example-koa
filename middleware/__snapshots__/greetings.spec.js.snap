// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greetings works complete 1`] = `
Object {
  "body": "Hello.",
  "response": Object {
    "set": [Function],
  },
}
`;

exports[`greetings works complete 2`] = `
Object {
  "body": "Hello. Remember to subscribe.",
  "response": Object {
    "set": [Function],
  },
}
`;

exports[`greetings works complete 3`] = `
Array [
  Array [
    "Etag",
    1234,
  ],
]
`;
